Domains/Jobs/Interfaces/JobRepositoryInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @file Domains/Jobs/Interfaces/JobRepositoryInterface.php
+ *
+ * Copyright (c) 2014-2022 Simon Fraser University
+ * Copyright (c) 2000-2022 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class JobRepositoryInterface
+ * @ingroup support
+ *
+ * @brief Interface for Domain's Jobs Repository
+ */
+
+namespace PKP\Domains\Jobs\Interfaces;
+
+use PKP\Support\Interfaces\Core\Repository;
+
+interface JobRepositoryInterface extends Repository
```

Comment:
Is this interface still necessary now that it's empty?

Reply:
This is necessary to enable the dependency injection of repository.

https://github.com/pkp/pkp-lib/pull/7553/files#diff-fbd6403657ede20e86a0f309dcf1a064e0da1d8748e4e65c2635ccfd8bc8d1deR29

Reply:
Is there a benefit to using the `JobRepositoryInterface` rather than `PKP\Domains\Jobs\Repositories\Job::class` in registering the dependency injection?

Reply:
Binding interface instead concrete class increase the malleability of system. Allowing us to implement more services with less effort (e.g moving the local database jobs table to SQS or RabbitMQ). Some comments about this subject, [here](https://stackoverflow.com/a/64589363/1890634).

The only 'mea culpa' I should mentioned is I've called the `PKP\Domains\Jobs\Repositories\Job::class` directly on the `Repo.inc.php` instead the Interface. :man_facepalming:

Reply:
Facilitating testing is a good argument for interfaces rather than classes in DI, so in principle I'm sold. However, rolling this out consistently would involve creating a *lot* of empty interfaces without an existing use case (e.g. tests) to justify them. Codebase bloat and promising but incompletely applied patterns are constant challenges. So I think it's OK to go ahead with this as is, but generally I'd prefer to just stick with `Job::class` until there's a specific need for an interface.

Reply:
Implement a better code coverage using unit tests is a wish I have. Those changes here are just the initial implementation on this subject. ;)